plugin-taskbar/ilxqttaskbarabstractmodel.h

```diff
@@ -0,0 +1,77 @@
+#ifndef ILXQTTASKBARABSTRACTMODEL_H
+#define ILXQTTASKBARABSTRACTMODEL_H
+
+#include <QAbstractListModel>
+
+//TODO: is this header public? Needed for WId type
+#include <qwindowdefs.h>
+
+//TODO: forward declare NET::Properties
+#include <netwm_def.h>
+
+//TODO: split in abstract interface and X11 implementation with KWindowSystem
+
+class LXQtTaskbarModelGroup
+{
+public:
+    LXQtTaskbarModelGroup() {}
+};
+
+class LXQtTaskbarModelWindow
+{
+public:
+    LXQtTaskbarModelWindow() {}
+
+    WId windowId;
+};
+
+class ILXQtTaskbarAbstractModel : public QAbstractListModel
+{
+    Q_OBJECT
+
+public:
+    explicit ILXQtTaskbarAbstractModel(QObject *parent = nullptr);
+
+    bool raiseOnCurrentDesktop() const;
+    void setRaiseOnCurrentDesktop(bool newRaiseOnCurrentDesktop);
+
+    bool showOnlyOneDesktopTasks() const;
+    void setShowOnlyOneDesktopTasks(bool newShowOnlyOneDesktopTasks);
+
+    int showDesktopNum() const;
+    void setShowDesktopNum(int newShowDesktopNum);
+
+    bool showOnlyCurrentScreenTasks() const;
+    void setShowOnlyCurrentScreenTasks(bool newShowOnlyCurrentScreenTasks);
+
+    bool showOnlyMinimizedTasks() const;
+    void setShowOnlyMinimizedTasks(bool newShowOnlyMinimizedTasks);
+
+    bool groupingEnabled() const;
+    void setGroupingEnabled(bool newGroupingEnabled);
+
+    bool ungroupedNextToExisting() const;
+    void setUngroupedNextToExisting(bool newUngroupedNextToExisting);
+
+    bool iconByClass() const;
+    void setIconByClass(bool newIconByClass);
+
+signals:
+    void refreshVisibility();
+    void iconChanged(WId windowId);
+    void urgencyChanged(WId windowId, bool urgency);
+    void titleChanged(WId windowId);
+
+private:
+    // Settings
+    bool mRaiseOnCurrentDesktop;
```

Comment:
Is this really a model? Model should be unaware of the view settings. So just a "plain" list of window information. The view should be responsible for showing particular information based on its configuration.
So IMO our windows list model should be an application singleton object and particular presentation widgets should be views using the model by their means.

Reply:
Yes but I dont't think it's UI code either.
Or maybe we should put a generic raise function on model and UI passes current desktop number or other desktop?

Reply:
I wasn't talking just about this {{mRaiseOnCurrentDesktop}}, but all of those. The model should be universal object storing and providing information for views.
The real information should be stored in one singleton model object and maybe each taskbar plugin can have it's proxy model (with source model being the singleton model one). Does this make sense?

Reply:
I think proxy model make things complex, especialli if you want one taskbar with grouping and another without it but both take from same model